utensor_cgen/frontend/tflite.py

```diff
@@ -0,0 +1,389 @@
+tensor_np_type[7] = np.dtype("int16")
+tensor_np_type[8] = np.dtype("cdouble")
+tensor_np_type[9] = np.dtype("int8")
+
+
+builtin_ops = {v: k for k, v in BuiltinOperator.__dict__.items()}
+
+
+def class_option2str(obj, idx):
+  names_lookup = {v: k for k, v in obj.__dict__.items()}
+  name = names_lookup[idx]
+  return str(idx) + " (" + name + ")"
+
+
+customOptionFormat_lookup = {v: k for k, v in CustomOptionsFormat.__dict__.items()}
+
+
+def fully_connected_op_data(op):
+  option_dict = {}
+  if op.CustomOptionsLength() < 1:
+    from .tflite_flatbuffer.FullyConnectedOptions import FullyConnectedOptions
+
+    option = FullyConnectedOptions()
+    builtin_data = op.BuiltinOptions()
+    option.Init(builtin_data.Bytes, builtin_data.Pos)
+    option_dict["FusedActivationFunction"] = class_option2str(
+      ActivationFunctionType, option.FusedActivationFunction()
+    )
+    option_dict["w_formats"] = class_option2str(
+      FullyConnectedOptionsWeightsFormat, option.WeightsFormat()
+    )
+  else:
+    option_dict[
+      customOptionFormat_lookup[op.CustomOptionsFormat()]
+    ] = op.CustomOptionsAsNumpy()
+  return option_dict
+
+
+def depthwise_conv2d_op_data(op):
+  option_dict = {}
+  if op.CustomOptionsLength() < 1:
+    from .tflite_flatbuffer.DepthwiseConv2DOptions import DepthwiseConv2DOptions
+
+    option = DepthwiseConv2DOptions()
+    builtin_data = op.BuiltinOptions()
+    option.Init(builtin_data.Bytes, builtin_data.Pos)
+    option_dict["Padding"] = option.Padding()
+    option_dict["StrideW"] = option.StrideW()
+    option_dict["StrideH"] = option.StrideH()
+    option_dict["DepthMultiplier"] = option.DepthMultiplier()
+    option_dict["FusedActivationFunction"] = class_option2str(
+      ActivationFunctionType, option.FusedActivationFunction()
+    )
+    option_dict["DilationWFactor"] = option.DilationWFactor()
+    option_dict["DilationHFactor"] = option.DilationHFactor()
+
+  else:
+    option_dict[
+      customOptionFormat_lookup[op.CustomOptionsFormat()]
+    ] = op.CustomOptionsAsNumpy()
+
+  return option_dict
+
+
+def reshape_op_data(op):
+  option_dict = {}
+  if op.CustomOptionsLength() < 1:
+    from .tflite_flatbuffer.ReshapeOptions import ReshapeOptions
+
+    option = ReshapeOptions()
+    builtin_data = op.BuiltinOptions()
+    option.Init(builtin_data.Bytes, builtin_data.Pos)
+    option_dict["NewShape"] = option.NewShapeAsNumpy()
+  else:
+    option_dict[
+      customOptionFormat_lookup[op.CustomOptionsFormat()]
+    ] = op.CustomOptionsAsNumpy()
+
+  return option_dict
+
+
+def dequantize_op_data(op):
+  if op.CustomOptionsLength() < 1:
+    from .tflite_flatbuffer.DequantizeOptions import DequantizeOptions
+
+    option = DequantizeOptions()
+    builtin_data = op.BuiltinOptions()
+    if builtin_data is None:
+      return builtin_data
+    option.Init(builtin_data.Bytes, builtin_data.Pos)
+    return [option]
+  else:
+    custom_option = op.CustomOptionsAsNumpy()
+    return [custom_option]
```

Comment:
Inconsistent return types.
Other handling functions return dictionaries, but `dequantize_op_data` is returning a list.
According to your parser's parsing logic, it looks like a bug.